src/kernel/arch/aarch64/tty.zig

```diff
@@ -24,51 +24,76 @@ const Framebuffer = struct {
 };
 
 const Pixel = packed struct {
-    red: u8,
-    green: u8,
     blue: u8,
+    green: u8,
+    red: u8,
     alpha: u8 = 0,
 };
 
 const font = [_][]const u1{
     &[_]u1{},
+    // space
+    &[_]u1{
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
+    },
     // !
     &[_]u1{
-        0, 0, 0, 1, 1, 0, 0, 0,
-        0, 0, 0, 1, 1, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 1, 1, 0, 0, 0,
         0, 0, 0, 1, 1, 0, 0, 0,
         0, 0, 0, 1, 1, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 1, 1, 0, 0, 0,
         0, 0, 0, 1, 1, 0, 0, 0,
+        0, 0, 0, 0, 0, 0, 0, 0,
     },
 };
 
 var framebuffer: Framebuffer = undefined;
 
 fn writePixel(x: usize, y: usize, pixel: Pixel) void {
-    //  log.debug("Writing pixel {}\n", .{pixel});
-    //  log.debug("    to ({}, {}) at fb 0x{x}, which is address {x}\n", .{ x, y, @ptrToInt(framebuffer.buffer), @ptrToInt(&framebuffer.buffer[y * framebuffer.bytes_per_row + x]) });
-    framebuffer.buffer[y * framebuffer.bytes_per_row + x] = pixel;
+    framebuffer.buffer[y * framebuffer.bytes_per_row / 4 + x] = pixel;
+}
+
+fn clearScreen() void {
+    var y: usize = 0;
+    while (y < framebuffer.height) : (y += 1) {
+        var x: usize = 0;
+        while (x < framebuffer.width) : (x += 1) {
+            writePixel(x, y, BLACK);
+        }
+    }
 }
 
 fn writeChar(x: usize, y: usize, char: u8) void {
     var ch = char;
     if (char < ' ' or char > '~')
         ch = ' ';
     // const bitmap = font[ch - ' '];
-    const bitmap = font[0];
+    var font_index: usize = undefined;
+    if (ch == ' ') {
+        font_index = 1;
+    } else {
+        font_index = 2;
+    }
+    const bitmap = font[font_index];
```

Comment:
Since the font array is ordered by ASCII, you can get the index by subtracting `ch` by 32.

Reply:
Done